cylc/flow/scripts/validate_reinstall.py

```diff
@@ -0,0 +1,154 @@
+)
+from cylc.flow.option_parsers import (
+    WORKFLOW_ID_ARG_DOC,
+    CylcOptionParser as COP,
+    combine_options,
+    log_subcommand,
+    cleanup_sysargv
+)
+from cylc.flow.id_cli import parse_id
+from cylc.flow.terminal import cli_function
+from cylc.flow.workflow_files import detect_old_contact_file
+
+
+CYLC_ROSE_OPTIONS = COP.get_cylc_rose_options()
+VR_OPTIONS = combine_options(
+    VALIDATE_OPTIONS,
+    REINSTALL_CYLC_ROSE_OPTIONS,
+    PLAY_OPTIONS,
+    CYLC_ROSE_OPTIONS,
+    modify={'cylc-rose': 'validate, install'}
+)
+
+
+def get_option_parser() -> COP:
+    parser = COP(
+        __doc__,
+        comms=True,
+        jset=True,
+        argdoc=[WORKFLOW_ID_ARG_DOC],
+    )
+    for option in VR_OPTIONS:
+        parser.add_option(*option.args, **option.kwargs)
+    return parser
+
+
+@cli_function(get_option_parser)
+def main(parser: COP, options: 'Values', workflow_id: str):
+    sys.exit(vro_cli(parser, options, workflow_id))
+
+
+def vro_cli(parser: COP, options: 'Values', workflow_id: str):
+    """Run Cylc (re)validate - reinstall - reload in sequence."""
+    # Attempt to work out whether the workflow is running.
+    # We are trying to avoid reinstalling then subsequently being
+    # unable to play or reload because we cannot identify workflow state.
+    workflow_id, *_ = parse_id(
+        workflow_id,
+        constraint='workflows',
+    )
+
+    # Use this interface instead of scan, because it can have an ambiguous
+    # outcome which we want to capture before we install.
+    try:
+        detect_old_contact_file(workflow_id)
+    except ServiceFileError:
+        # Workflow is definately running:
+        workflow_running = True
+    else:
+        # Workflow is definately stopped:
+        workflow_running = False
+
+    # Force on the against_source option:
+    options.against_source = True   # Make validate check against source.
+    log_subcommand('validate --against-source', workflow_id)
+    cylc_validate(parser, options, workflow_id)
+
+    log_subcommand('reinstall', workflow_id)
+    reinstall_ok = cylc_reinstall(options, workflow_id)
+    if not reinstall_ok:
+        LOG.warning(
+            'No changes to source: No reinstall or'
+            f' {"reload" if workflow_running else "play"} required.'
+        )
+        return 1
+
+    # Run reload if workflow is running, else play:
+    if workflow_running:
+        log_subcommand('reload', workflow_id)
+        cylc_reload(options, workflow_id)
+
+    # run play anyway, to resume a paused workflow:
+    else:
+        cleanup_sysargv(
+            'play',
+            workflow_id,
+            options,
+            compound_script_opts=VR_OPTIONS,
+            script_opts=(
+                PLAY_OPTIONS + CYLC_ROSE_OPTIONS
+                + parser.get_std_options()
+            ),
+            source='',  # Intentionally blank
+        )
+        log_subcommand('play', workflow_id)
+        scheduler_cli(options, workflow_id)
```

Comment:
I think we might need a `return 0` at the end of this method?

Reply:
Actually I think this is fine, turns out `sys.exit(None)` is equivalent to `sys.exit(0)`.